nomad/grapl-core.nomad

```diff
@@ -0,0 +1,428 @@
+        task "graph-merger" {
+            driver = "docker"
+
+            config {
+                image = "${var.container_registry}/graph-merger:${var.tag}"
+                args = ["/graph-merger"]
+            }
+
+            env {
+                REDIS_ENDPOINT = "${var.redis_endpoint}"
+                MG_ALPHAS = "${local.alpha_grpc_connect_str}"
+                GRAPL_SCHEMA_TABLE = "${var.schema_table_name}"
+                AWS_REGION = "${var.aws_region}"
+                # https://github.com/grapl-security/grapl/blob/18b229e824fae99fa2d600750dd3b17387611ef4/pulumi/grapl/__main__.py#L165
+                DEST_BUCKET_NAME = "subgraphs-merged-bucket"
+                SOURCE_QUEUE_URL = "${var.aws_sqs_url}/${var.aws_account_id}/graph-merger-queue"
+                DEAD_LETTER_QUEUE_URL = "${var.aws_sqs_url}/${var.aws_account_id}/graph-merger-dead-letter-queue"
+            }
+        }
+
+        service {
+            connect {
+                sidecar_service {
+                    proxy {
+                        dynamic "upstreams" {
+                            iterator = alpha
+                            for_each = local.dgraph_alphas
+
+                            content {
+                                destination_name = "dgraph-alpha-${alpha.value.id}-grpc-public"
+                                local_bind_port = alpha.value.grpc_public_port
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    group "grapl-node-identifier" {
+        count = var.node_identifiers
+
+        network {
+            mode = "bridge"
+        }
+
+        task "node-identifier" {
+            driver = "docker"
+
+            config {
+                image = "${var.container_registry}/node-identifier:${var.tag}"
+                args = ["/node-identifier"]
+            }
+
+            env {
+                REDIS_ENDPOINT = "${var.redis_endpoint}"
+                MG_ALPHAS = "${local.alpha_grpc_connect_str}"
+                GRAPL_SCHEMA_TABLE = "${var.schema_table_name}"
+                AWS_REGION = "${var.aws_region}"
+                DYNAMIC_SESSION_TABLE = "${var.session_table}"
+                # https://github.com/grapl-security/grapl/blob/18b229e824fae99fa2d600750dd3b17387611ef4/pulumi/grapl/__main__.py#L156
+                DEST_BUCKET_NAME = "subgraphs-generated-bucket"
+                SOURCE_QUEUE_URL = "${var.aws_sqs_url}/${var.aws_account_id}/node-identifier-queue"
+                DEAD_LETTER_QUEUE_URL = "${var.aws_sqs_url}/${var.aws_account_id}/node-identifier-dead-letter-queue"
+            }
+        }
+    }
+
+    group "grapl-node-identifier-retry" {
+        count = var.node_identifiers_retry
+
+        network {
+            mode = "bridge"
+        }
+
+        task "node-identifier-retry" {
+            driver = "docker"
+
+            config {
+                image = "${var.container_registry}/node-identifier-retry:${var.tag}"
```

Comment:
FYI, with the way we're managing our containers and pinning their versions in the release pipeline, there is no guarantee that all the services' tags will be the same (I think for the time being they are by default, but this will not always be the case).

Reply:
Gotcha; so we'll need a tag-per-service, then?

Reply:
Eventually, yes.

Reply:
I'll add tags for the graph merger and node identifier. I feel like it's safe to assume that node-identifier and node-identifier-retry will share a tag. If not, let me know!

Reply:
Yes, I believe with how things are set up now, node-identifer and node-identifier-retry will always have the same release tag.